buildlog/coder.go

```diff
@@ -105,11 +105,12 @@ func newAgentClientV2(ctx context.Context, logger slog.Logger, client *agentsdk.
 	var conn drpc.Conn
 	var err error
 	for r := retry.New(10*time.Millisecond, time.Second); r.Wait(ctx); {
-		conn, err = client.ConnectRPC(ctx)
+		c, err := client.ConnectRPC20(ctx)
 		if err != nil {
 			logger.Error(ctx, "connect err", slog.Error(err))
 			continue
 		}
+		conn = c.DRPCConn()
```

Comment:
`client.ConnectRPC20` returns the equivalent to the `arpc` below.  Grabbing the underlying `Conn` and then recreating the RPC client isn't necessary and breaks the type-checking that should prevent you from using RPCs that aren't present in v2.0 of the Agent API